Sources/NIOCore/AsyncAwaitSupport.swift

```diff
@@ -62,8 +63,11 @@ extension EventLoopPromise {
     /// - returns: A `Task` which was created to `await` the `body`.
     @available(macOS 10.15, iOS 13, tvOS 13, watchOS 6, *)
     @discardableResult
+    @preconcurrency
     @inlinable
-    public func completeWithTask(_ body: @escaping @Sendable () async throws -> Value) -> Task<Void, Never> {
+    public func completeWithTask(
+        _ body: @escaping @Sendable () async throws -> Value
```

Comment:
Pretty sure this is incorrect. We're running `body` in one place and it can return a non-Sendable `Value` which we then transfer onto the EL. Nothing restricts `Value` from still being accessed on the previous place.

Reply:
I agree with Franz, this seems fine. So far as I can see, we can only close over `Sendable` state in the closure, so anything we closed over must be safe to send. Assuming that's true, any value we return was either closed over, or produced on the EL (as that's where the body executes), so I think this is fine.

Reply:
I missed the `where Value: Sendable`, so this seems fine. I assumed that it can return a non-Sendable Value because I missed it but that's not the case.